### PR TITLE
Update JS Docs

### LINKED ISSUE #543: Documentation for Aggregate Queries
Someone noted in [#4197](https://github.com/parse-community/parse-server/issues/4197) on the parse server repo that we do not have documentation for aggregate queries in the docs. I'm opening this up to make this issue known until we get proper documentation in place for this functionality.

### PR DESCRIPTION
**Pull request by dplewis**
Closes: https://github.com/parse-community/Parse-SDK-JS/issues/543

* Polygon
* Schema API
* Aggregate and Distinct
* Full Text

I'm not the best at explaining so any help would be great.

Also If there are any new JS features I missed I can add them here.

I'll use this as a base for iOS and Android Guides

### CODE REVIEW

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: I would put the note regarding the 'cost' of full text search ops up here instead. That way it's a disclaimer of sorts as to what kind of impact this may have, before they start seeing any examples.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: *this would be the note to move up ^, and maybe with some further clarification as to 'expensive operations'.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Should either rename to 'pipelineArray' or change from **const** to **var** or **let**. In the same example you have two constants declared with the same name, and although it's just an example it should probably show correct code 😉 .

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: typo:
> Most operation**s** in...

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: I would add some descriptive text before this code block to detail what's in it, especially since it's only preceded by another block.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Same thing here, something before it to describe and break it up would be nice.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Missing a period (or colon) at the end of this sentence

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md]: typo on plural column:
> to display column**~~s~~** names and

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: So although Parse-Server is accurate to the project repo name we should probably be describing it as **Parse Server**. Unless we're referring to the repo itself the spaced name is a more appropriate way to describe it in most cases.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Parse-Server thing

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Parse-Server thing

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md]: Parse-Server thing

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md]: could put a space here to visually separate default from custom fields, purely stylistic however.

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md:62]: Probably would be good to add a little warning with this as well.
> With great power comes great responsibility. Altering the schema directly should be done with care, you can't go back to retrieve data if you remove a field and it's associated values.

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md]: Could italicize *be careful* to stress it.

File: _includes/js/schemas.md
Comment (montymxb) [inline _includes/js/schemas.md]: typo, extra s:
> display column**~~s~~** names

File: _includes/js/schemas.md
Comment (montymxb) [inline _includes/js/schemas.md]: Parse-Server thing

File: _includes/rest/schemas.md
Comment (montymxb) [inline _includes/rest/schemas.md]: Parse-Server thing

File: _includes/js/schemas.md
Comment (montymxb) [inline _includes/js/schemas.md]: Wait a minute...this is a duplicate file of **schema.md** right? Is there a particular reason why this one (or that one) is being added?

File: assets/js/bundle.js
Comment (montymxb) [inline assets/js/bundle.js]: Reason for this?

File: js.md
Comment (montymxb) [inline js.md:25]: If you're just adding **schema.md** you might want to drop **schemas.md**. In the php section I have it setup to reference schema.md already, might as well be consistent.

**Review by montymxb — CHANGES_REQUESTED** [review]
Various changes, questions, clarifications and such. There's probably some other stuff that will need to be caught in a second review once this initial items are addressed.

File: _includes/js/schema.md
Comment (dplewis) [reply _includes/js/schema.md:62]: I like

File: _includes/js/schemas.md
Comment (dplewis) [reply _includes/js/schemas.md]: Oops

File: assets/js/bundle.js
Comment (dplewis) [reply assets/js/bundle.js]: No Idea, I think when I ran `bundle install` this happened or maybe my text-editor linter.

I'll change it back

**Comment by dplewis** [comment]
@montymxb Thanks for the initial review. I went ahead and added some more aggregate examples.

File: _includes/js/queries.md
Comment (dplewis) [reply _includes/js/queries.md]: I've typed parse-server more than I've typed my own name

File: assets/js/bundle.js
Comment (montymxb) [reply assets/js/bundle.js]: That's fine then, so as it's functional.

**Comment by dplewis** [comment]
@flovilmart Can you look over this?

**Review by flovilmart — COMMENTED** [review]
I agree with @montymxb comments, otherwise LGTM

File: _includes/rest/schemas.md
Comment (montymxb) [inline _includes/rest/schemas.md:12]: Didn't even realize the 3.0 thing there...nice catch

File: _includes/js/geopoints.md
Comment (montymxb) [inline _includes/js/geopoints.md]: can probably replace the / with 'or' just to be a bit more explicit.

File: _includes/js/geopoints.md
Comment (montymxb) [inline _includes/js/geopoints.md]: Stylistic
> You can **also** query for whether...

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Stylistic
> **You can** use `fullText` for efficient...

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Stylistic
> * Note: Full Text Search can be resource **intensive**. Ensure the cost of using indexes is worth the benefit, see [storage requirements & performance costs of text indexes](https://docs.mongodb.com/manual/core/index-text/#storage-requirements-and-performance-costs).

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: I was wondering about these requirement labels, we should try to standardize them so they're not just dotted around. ~~I think a solution could be to make sure this immediately follows the feature title. In this case `### Full Text Search`. Also, and this is just a thought, but maybe we should italicize this entire requirement, just to emphasize it before the reader attempts to start implementing said feature.~~

Make sure to lead the requirement with `*` immediately following the introductory paragraph of the feature.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: Same thing here, we should move this statement directly below the title of `Aggregate`. Should make sure it's similar to the others as well. We can keep the master key requirement in there though, makes sense to have that as well.

Comment (montymxb) [reply _includes/js/queries.md]: ~~Also I'm not set in stone on this one. Using an `*` marks it with a blue + on the docs, and that might be good as is, but location should be standardized below the title.~~

Positioning is good here, but you could break up the masterKey & server requirement, with the server one first.

File: _includes/js/queries.md
Comment (montymxb) [inline _includes/js/queries.md]: ~~same here~~
Could break it up.

File: _includes/js/schema.md
Comment (montymxb) [inline _includes/js/schema.md]: ~~same here.~~
Positioning is pretty good here, and the parts are broken up. This how the others should be done as well. Could also stand to flip the order so the server requirement is prioritized before the master key.

**Review by montymxb — CHANGES_REQUESTED** [review]
Alright second look over and it's pretty good! Just some nits here and there.

To note I'm not 100% sure I want to move those locations for server requirements, as I see they're already following the leading paragraph in other sections of the docs. To retract what I put down before, I would use that existing style instead, that and the blue note is fine (and don't worry about the italics). I'll update those comments shortly to reflect this.

**Comment by montymxb** [comment]
Just amended some of those comments in the review.

**Comment by dplewis** [comment]
Thanks again, changes made

**Review by montymxb — APPROVED** [review]
Looks approved :octocat:

**Comment by montymxb** [comment]
Thanks again for putting up the PR @dplewis . Guess we'll be seeing a couple for iOS and Android pretty soon too!

**Pull request merged**
